convert2rhel/actions/efi.py

```diff
@@ -0,0 +1,78 @@
+# Copyright(C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+import os.path
+
+from convert2rhel import __version__ as installed_convert2rhel_version
+from convert2rhel import actions, grub, utils
+from convert2rhel.systeminfo import system_info
+
+
+logger = logging.getLogger(__name__)
+
+
+class Efi(actions.Action):
+    id = "EFI"
+    dependencies = tuple()
+
+    def run(self):
+        """Inhibit the conversion when we are not able to handle UEFI."""
+        logger.task("Prepare: Check the firmware interface type (BIOS/UEFI)")
+        if not grub.is_efi():
+            logger.info("BIOS detected.")
+            return
+        logger.info("UEFI detected.")
+        if not os.path.exists("/usr/sbin/efibootmgr"):
+            self.status = actions.STATUS_CODE["ERROR"]
+            self.error_id = "EFIBOOTMGR_NOT_FOUND"
+            self.message = "Install efibootmgr to continue converting the UEFI-based system."
```

Comment:
This can be change to use `self.set_result()` when https://github.com/oamg/convert2rhel/pull/767 is merged.